src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs

```diff
@@ -157,6 +174,22 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         {
             if (eventSource.Name.StartsWith(EventSourceNamePrefix, StringComparison.Ordinal))
             {
+                // If there are EventSource classes already initialized as of now, this method would be called from
```

Comment:
https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/WEB/Src/EventCounterCollector/EventCounterCollector/EventCounterListener.cs#L28 similar/same issue in other places. 
Don't recollect if you need to use a Concurrent data structure, or not. (probably no need as an explicit lock is used?)

Reply:
I think it's not needed because it's already locked. Either concurrent data structure or a lock works. There could be a performance difference, but I don't think we will have enough number of EventSource sub-classes/instances for the difference to matter.